src/pyrokinetics/diagnostics/synthetic_highk_dbs.py

```diff
@@ -0,0 +1,670 @@
+        Rloc: float,  # [1.26] [m]      Bhavin 47107: 1.2689; David 22769: R=1.086, 1,137
+        Zloc: float,  # [m]        Bhavin 47107: 0.1692; David 22769: 0.18
+        Kn0_exp: ArrayLike,  # np.asarray([21.637153 ]) # np.asarray([-21.637153])  #np.asarray([0, 0])       # [cm-1]   usually 0 for DBS, finite for high-k
+        Kb0_exp: ArrayLike,  # np.asarray([2.701665 ])   # np.asarray([-2.701665])  # np.asarray([1.75, 6.903])       # [cm-1], Bhavin 47107: 6.903
+        wR: float,  # [m]    local sim: do sinc function
+        wZ: float,  # 2/1711.94563       # [m]    wZ 0.02 MAST-U
+        eq_file: PathLike,  #
+        kinetics_file: PathLike,  #
+        simdir: PathLike,  #
+        savedir: PathLike,  #
+        fsize: int,
+    ):
+        self.diag = diag
+        self.syn_filter = syn_filter
+        self.Rloc = Rloc
+        self.Zloc = Zloc
+        self.Kn0_exp = Kn0_exp
+        self.Kb0_exp = Kb0_exp
+        self.wR = wR
+        self.wZ = wZ
+        self.eq_file = eq_file
+        self.kinetics_file = kinetics_file
+        self.simdir = simdir
+        self.savedir = savedir
+        self.fsize = fsize
+
+        self.Pkf = np.zeros(np.size(Kn0_exp))
+        self.Pkf_hann = np.zeros(np.size(Kn0_exp))
+        self.Pkf_kx0ky0 = np.zeros(np.size(Kn0_exp))
+        self.Pks = np.zeros(np.size(Kn0_exp))
+        self.Sigma_ks_hann = np.zeros(np.size(Kn0_exp))
+
+        # calcualte thetaloc
+        pyro = Pyro(
+            eq_file=eq_file,
+            kinetics_file=kinetics_file,
+            gk_file=simdir + "/input.cgyro",
+        )
+        self.pyro = pyro
+        self.eq = pk.read_equilibrium(eq_file)
+        meter = pyro.norms.units.meter
+        self.psin = self.eq._psi_RZ_spline(Rloc * meter, Zloc * meter) / (
+            self.eq.psi_lcfs - self.eq.psi_axis
+        )
+        pyro.load_local(psi_n=self.psin, local_geometry="Miller")
+        self.geo = pyro.local_geometry
+        pyro.load_metric_terms()
+        self.m = pyro.metric_terms
+
+        # normalizations
+        [
+            self.a_minor,
+            self.te_ref,
+            self.ne_ref,
+            self.csa,
+            self.rhos_unit,
+        ] = self.get_units_norms(pyro)
+        Qgbunit = (
+            self.ne_ref * self.csa * self.a_minor * self.te_ref * (self.rhos_unit) ** 2
+        )  # [W/m^2]
+        print("QgB,unit = " + str(np.round(Qgbunit / 1e3, 4)) + " [kW/m^2]")
+
+        # find thetaloc:
+        thetatmp = self.geo.theta[self.geo.Z > self.geo.Z0]
+        Rtmp = self.geo.R[self.geo.Z > self.geo.Z0] * self.a_minor  # [m]
+        Ztmp = self.geo.Z[self.geo.Z > self.geo.Z0] * self.a_minor  # [m]
+        tmp_ind = np.argmin(np.abs(Rtmp - Rloc))
+        self.thetaloc = thetatmp[tmp_ind]  # np.interp(Zloc, Ztmp, thetatmp)
+
+        print("psin = " + str(self.psin))
+        print("theta/pi = " + str(self.thetaloc / np.pi))
+        print("Rloc = " + str(Rloc) + " [m]")
+        print("Zloc = " + str(Zloc) + " [m]")
```

Comment:
Rather than calling `print` here, add a `__repr__` method (for a "plain" representation that could recreate the object) or `__str__` (for a "nicer" representation).

Also, rather than `"string" + str(something)` use an f-string:

```suggestion
        print(f"psin = {self.psin}")
        print(f"theta/pi = {self.thetaloc / np.pi}")
        print(f"Rloc = {Rloc} [m]")
        print(f"Zloc = {Zloc} [m]")
```